src/machines/java/com/enderio/machines/common/item/FluidTankItem.java

```diff
@@ -26,4 +55,67 @@ public BlockEntityWithoutLevelRenderer getCustomRenderer() {
             }
         });
     }
+
+    @Override
+    public void addCommonTooltips(ItemStack itemStack, @Nullable Player player, List<Component> tooltips) {
+        itemStack.getCapability(ForgeCapabilities.FLUID_HANDLER_ITEM).ifPresent(iFluidHandlerItem -> {
+            if (iFluidHandlerItem instanceof FluidItemStack fluidHandler) {
+                if (fluidHandler.getFluid().isEmpty()) {
+                    tooltips.add(TooltipUtil.style(EIOLang.TANK_EMPTY_STRING));
+                } else {
+                    tooltips.add(TooltipUtil.styledWithArgs(EIOLang.FLUID_TANK_TOOLTIP, fluidHandler.getFluid().getAmount(), capacity,
+                        fluidHandler.getFluid().getFluid().getFluidType().getDescription().getString()));
+                }
+            }
+        });
+    }
+
+    @Nullable
+    @Override
+    public ICapabilityProvider initCapabilities(ItemStack stack, @Nullable CompoundTag nbt) {
+        return new FluidItemStack(stack, capacity);
+    }
+
+    public class FluidItemStack extends FluidHandlerItemStack {
+        private static final String BLOCK_ENTITY_TAG = "BlockEntityTag";
+
+        /**
+         * @param container The container itemStack, data is stored on it directly as NBT.
+         * @param capacity  The maximum capacity of this fluid tank.
+         */
+        public FluidItemStack(@NotNull ItemStack container, int capacity) {
+            super(container, capacity);
+        }
+
+        @NotNull
+        public FluidStack getFluid() {
+            Optional<CompoundTag> tagCompoundOptional = Optional.ofNullable(container.getTag());
+            return tagCompoundOptional
+                .map(tagCompound -> tagCompound.getCompound(BLOCK_ENTITY_TAG))
+                .map(beTag -> beTag.getCompound(FluidTankBlockEntity.FLUID_TAG_KEY))
+                .map(FluidStack::loadFluidStackFromNBT)
+                .orElse(FluidStack.EMPTY);
+        }
+
+        protected void setFluid(FluidStack fluid) {
+            CompoundTag mainTag = container.getOrCreateTag();
+            CompoundTag beTag = new CompoundTag();
+            mainTag.put(BLOCK_ENTITY_TAG, beTag);
+
+            CompoundTag fluidTag = new CompoundTag();
+            fluid.writeToNBT(fluidTag);//rewrites the old value
+            //TODO: externalize FluidTankBlockEntity.FLUID_TAG_KEY into one global parameter.
+            beTag.put(FluidTankBlockEntity.FLUID_TAG_KEY, fluidTag);
+        }
+
+        @Override
+        protected void setContainerToEmpty() {
+            CompoundTag tagCompound = container.getTag();
+            if (tagCompound != null) {
+                CompoundTag BETag = tagCompound.getCompound(BLOCK_ENTITY_TAG);
+                if (BETag.contains(FluidTankBlockEntity.FLUID_TAG_KEY))
+                    BETag.remove(FluidTankBlockEntity.FLUID_TAG_KEY);
```

Comment:
```suggestion
                CompoundTag blockEntityTag = tagCompound.getCompound(BLOCK_ENTITY_TAG);
                if (blockEntityTag.contains(FluidTankBlockEntity.FLUID_TAG_KEY))
                    blockEntityTag.remove(FluidTankBlockEntity.FLUID_TAG_KEY);
```
non constant variables don't start with a capital letter